internal/parser/simple_parser_rules.go

```diff
@@ -1228,3 +1232,1510 @@ func (p *simpleParser) parseCreateVirtualTableStmt(createToken token.Token, r re
+					next, ok = p.lookahead(r)
+					if !ok {
+						return
+					}
+					if next.Value() == "," {
+						p.consumeToken()
+					} else {
+						break
+					}
+				}
+			}
+		}
+
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		if next.Type() == token.KeywordWhere {
+			stmt.Where = next
+			p.consumeToken()
+			stmt.Expr1 = p.parseExpression(r)
+		}
+
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		if next.Type() == token.KeywordGroup {
+			stmt.Group = next
+			p.consumeToken()
+			next, ok = p.lookahead(r)
+			if !ok {
+				return
+			}
+			if next.Type() == token.KeywordBy {
+				stmt.By = next
+				p.consumeToken()
+			}
+			for {
+				stmt.Expr2 = append(stmt.Expr2, p.parseExpression(r))
+				next, ok = p.lookahead(r)
+				if !ok {
+					return
+				}
+				if next.Value() == "," {
+					p.consumeToken()
+				} else {
+					break
+				}
+			}
+
+			next, ok = p.lookahead(r)
+			if !ok {
+				return
+			}
+			if next.Type() == token.KeywordHaving {
+				stmt.Having = next
+				p.consumeToken()
+				stmt.Expr3 = p.parseExpression(r)
+			}
+		}
+
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		if next.Type() == token.KeywordWindow {
+			// Consuming the keyword window
+			stmt.Window = next
+			p.consumeToken()
+			for {
+				stmt.NamedWindow = append(stmt.NamedWindow, p.parseNamedWindow(r))
+
+				next, ok = p.optionalLookahead(r)
+				if !ok || next.Type() == token.EOF {
+					return
+				}
+				if next.Value() == "," {
+					p.consumeToken()
+				} else {
+					break
+				}
+			}
+		}
+	}
+	if next.Type() == token.KeywordValues {
+		stmt.Values = next
+		p.consumeToken()
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		if next.Value() == "(" {
+			for {
+				stmt.ParenthesizedExpressions = append(stmt.ParenthesizedExpressions, p.parseParenthesizeExpression(r))
+				next, ok = p.lookahead(r)
+				if !ok {
+					return
+				}
+				if next.Value() == "," {
+					p.consumeToken()
+				} else {
+					break
+				}
+			}
+		} else {
+			r.unexpectedSingleRuneToken('(')
+		}
+	}
+
+	// Checking whether there is a token that leads to a part of the statement
+	// ensures that stmt.CompoundOperator is nil, instead of an assigned empty value.
+	next, ok = p.optionalLookahead(r)
+	if !ok || next.Type() == token.EOF {
+		return
+	}
+	if next.Type() == token.KeywordUnion || next.Type() == token.KeywordIntersect || next.Type() == token.KeywordExcept {
+		stmt.CompoundOperator = p.parseCompoundOperator(r)
+	}
+	return
+}
+
+// parseResultColumn parses result-column as defined in:
+// https://sqlite.org/syntax/result-column.html
+func (p *simpleParser) parseResultColumn(r reporter) (stmt *ast.ResultColumn) {
+	stmt = &ast.ResultColumn{}
+	tableNameOrAsteriskOrExpr, ok := p.lookahead(r)
+	if !ok {
+		return
+	}
+	if tableNameOrAsteriskOrExpr.Value() == "*" {
+		stmt.Asterisk = tableNameOrAsteriskOrExpr
+		p.consumeToken()
+	} else if tableNameOrAsteriskOrExpr.Type() == token.Literal {
```

Comment:
Expression is only parsed if there's a literal?
Also, use a switch to differentiate between three cases, maybe use expr as default

Reply:
Yes, the parsing strictly happens when there's a literal.
This is one of the discussions we had in the past where I asked what I should do because the literal is consumed by the expression too.
What I'm doing is, if its a literal, it can be still an expr or the table name. So I wait it out, check for the period. 

I have missed a case where I parse where expr doesnt start with a literal, Ill add that.